db-service/lib/SQLService.js

```diff
@@ -79,7 +39,16 @@ class SQLService extends DatabaseService {
     let rows = await ps.all(values)
     if (rows.length)
       if (cqn.SELECT.expand) rows = rows.map(r => (typeof r._json_ === 'string' ? JSON.parse(r._json_) : r._json_ || r))
-    if (cqn.SELECT.count) rows.$count = await this.count(query, rows)
+    if (cqn.SELECT.count) rows.$count = await this.count(query, rows) 
+    if (cds.env.features.compat_stream_cqn) {
+      if (query._streaming) {
+        this._changeToStreams(cqn, rows, true)
+        return rows.length ? { value: Object.values(rows[0])[0] } : undefined
+      } 
+    } else {  
+      this._changeToStreams(cqn, rows)
```

Comment:
Calling `_changeToStreams` should not be done in the base `SQLService` as it is only implemented in `SQLite`.

Reply:
kind of solution implemented